<?xml version="1.0"?>

<launch>
    <arg name="log_path" default="$(find metacontrol_experiments)/data/"/>
    <arg name="record_bags" default="false"/>
    <arg name="bag_store_path" default="$(find metacontrol_experiments)/bags/"/>
    <arg name="obstacles" default="0"/>
    <arg name="goal_nr" default="1"/>

    <node pkg="metacontrol_experiments" type="stop_simulation_node" name="stop_simulation_node" output="screen" required="true">
        <param name="data_log_folder" value="$(arg log_path)"/>
        <param name="store_data_freq" value="1.0" type="double"/>
    </node>
    <!-- Record data  -->
    <node pkg="rosbag" type="record" name="record_bag_node" output="screen" if="$(arg record_bags)"
        args="-o $(arg bag_store_path)/metacontrol_
        /amcl_pose
        /power_load
        /odom
        /imu/data
        /d_obstacle
        /diagnostics
        /move_base/result
        /move_base/goal"
        required="true">
    </node>
    <node pkg="metacontrol_experiments" type="create_unknown_obstacles.py" name="create_unknown_obstacles_node" output="screen" required="true">
        <param name="goal_nr" value="$(arg goal_nr)" type="int"/>
        <param name="obstacles" value="$(arg goal_nr)" type="int"/>
    </node>


</launch>